crates/extensions/c8y_mapper_ext/src/availability/mod.rs

```diff
@@ -0,0 +1,62 @@
+use crate::availability::actor::TimerPayload;
+pub use builder::AvailabilityBuilder;
+use c8y_api::smartrest::topic::C8yTopic;
+use tedge_actors::fan_in_message_type;
+use tedge_api::entity_store::EntityExternalId;
+use tedge_api::entity_store::EntityRegistrationMessage;
+use tedge_api::mqtt_topics::MqttSchema;
+use tedge_api::mqtt_topics::ServiceTopicId;
+use tedge_api::HealthStatus;
+use tedge_config::TEdgeConfig;
+use tedge_config::TopicPrefix;
+use tedge_timer_ext::SetTimeout;
+use tedge_timer_ext::Timeout;
+
+mod actor;
+mod builder;
+#[cfg(test)]
+mod tests;
+
+pub type TimerStart = SetTimeout<TimerPayload>;
+pub type TimerComplete = Timeout<TimerPayload>;
+pub type SourceHealthStatus = (ServiceTopicId, HealthStatus);
+
+fan_in_message_type!(AvailabilityInput[EntityRegistrationMessage, SourceHealthStatus, TimerComplete] : Debug);
+fan_in_message_type!(AvailabilityOutput[C8ySmartRestSetInterval117, C8yJsonInventoryUpdate] : Debug);
+
+// TODO! Make it generic and move to c8y_api crate while refactoring c8y-mapper
```

Comment:
Better to do this right now.

This crate is already using `c8y_api::smartrest::inventory::set_required_availability_message` so moving `C8ySmartRestSetInterval117` there should be easy!

Reply:
Done in 2e05dc2b36fc1d10b9c6bcc28bd3e492e5520b05

Reply:
Done in 2e05dc2b36fc1d10b9c6bcc28bd3e492e5520b05